src/main/java/TulipTask.java

```diff
@@ -0,0 +1,111 @@
+//                System.out.println(splitStr[1]);
+//            }
+
+            String command;
+
+            if (input.toLowerCase().contains("mark")) {
+                String[] splitStr = input.split(" ");
+                command = splitStr[0];
+            } else {
+                command = input.toLowerCase();
+            }
+
+            switch(command) {
+                case "list":
+                    listTasks(list);
+                    break;
+
+                case "mark":
+                    String[] splitStr = input.split(" ");
+                    int index = Integer.parseInt(splitStr[1]) - 1;
+                    markAsDone(list.get(index));
+                    break;
+
+                case "unmark":
+                    String[] split = input.split(" ");
+                    int idx = Integer.parseInt(split[1]) - 1;
+                    markAsNotDone(list.get(idx));
+                    break;
+
+                case "bye":
+                    System.out.println("--------------------------------------------");
+                    System.out.println("Bye! Hope to see you again soon :)");
+                    System.out.println("--------------------------------------------");
+                    return;
+
+                default:
+                    Task task = new Task(command);
+                    list.add(task);
+                    echo(task);
+                    break;
+            }
+        }
+    }
+
+    public static void echo (Task task) {
+        System.out.println("--------------------------------------------");
+        System.out.println("added: " + task.description);
+        System.out.println("--------------------------------------------");
+    }
+
+    public static void listTasks (ArrayList<Task> list) {
+        System.out.println("--------------------------------------------");
+        System.out.println("Here are your tasks: ");
+        for (int i = 0; i < list.size(); i++) {
+            Task task = list.get(i);
+            System.out.println((i + 1) + ". " + "[" + task.getStatusIcon() + "] " + task.description);
+        }
+        System.out.println("--------------------------------------------");
+    }
+
+    public static void markAsDone(Task task) {
+        task.markAsDone();
+        System.out.println("--------------------------------------------");
+        System.out.println("Great job! I have marked this task as done: \n" + "  [" + task.getStatusIcon() + "] " + task.description);
+        System.out.println("--------------------------------------------");
+    }
+    public static void markAsNotDone(Task task) {
+        task.markAsNotDone();
+        System.out.println("--------------------------------------------");
+        System.out.println("Okay, I have marked this task as not done: \n" + "  [" + task.getStatusIcon() + "] " + task.description);
+        System.out.println("--------------------------------------------");
+    }
+}
```

Comment:
You may consider adding comment statements here to explain to other engineers what and why you are doing this.